core/controllers/admin.py

```diff
@@ -1008,3 +1009,68 @@ def put(self):
         user_services.log_username_change(
             self.user_id, old_username, new_username)
         self.render_json({})
+
+
+class NumberOfDeletionRequestsHandler(base.BaseHandler):
+    """Handler for deleting account via admin page."""
+
+    GET_HANDLER_ERROR_RETURN_TYPE = feconf.HANDLER_TYPE_JSON
+
+    @acl_decorators.can_access_admin_page
+    def get(self):
+        self.render_json({
+            'number_of_pending_deletion_models': (
+                wipeout_service.get_number_of_pending_deletion_requests())
+        })
+
+
+class DeleteAccountHandler(base.BaseHandler):
+    """Handler for deleting account via admin page."""
+
+    @acl_decorators.can_access_admin_page
+    def post(self):
+        email = self.payload.get('email', None)
+
+        if email is None:
+            raise self.InvalidInputException(
+                'Invalid request: email must be specified')
+
+        if not isinstance(email, python_utils.UNICODE):
+            raise self.InvalidInputException(
+                'Expected email to be a unicode string, received %s' % email)
+
+        pending_deletion_request = (
```

Comment:
I'm not sure I understand this part. Maybe add some additional details as comments?

In particular, if this handler kicks off a new deletion request, why would we expect a pending deletion request to be present?

Reply:
1. user requests deletion → the pending deletion request is created
2. either the repeating cronjob or this handler kicks off the deletion

Reply:
Done, removed.